### PR TITLE
Passing gRPC max msg size to proxy client stream

### PR DESCRIPTION
**Pull request by Flern**
# Description

Updated the `handler.go` file to use a passed-in argument of `maxMessageBodySizeMB` that is provided from the `runtime.go` instantiation of the proxy down through the `TransparentHandler` by using the configuration item `--dapr-http-max-request-size` parameter.  If there is no parameter configured, a zero value is propagated, so any value of zero (or less) is ignored and defaults are used.

## Issue reference

https://github.com/dapr/dapr/issues/6349

Please reference the issue this PR will close: #_6349_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by ItalyPaleAle** [comment]
Overall I think this is fine. 2 comments:

1. Using `dapr-http-max-request-size` doesn't feel right for something that is related to gRPC. Should we have a separate CLI flag for this maybe? Also, whatever flag should also apply to our gRPC server too perhaps (not just for gRPC proxying).
2. Could you please update one of the E2E tests, such as the gRPC proxying one, to validate that with this flag we can send messages larger than 4MB?

**Comment by yaron2** [comment]
> 1. Using `dapr-http-max-request-size` doesn't feel right for something that is related to gRPC. Should we have a separate CLI flag for this maybe? Also, whatever flag should also apply to our gRPC server too perhaps (not just for gRPC proxying).

We should reuse `dapr-http-max-request-size` here as this is already applied to our gRPC servers.

**Comment by ItalyPaleAle** [comment]
> > 1. Using `dapr-http-max-request-size` doesn't feel right for something that is related to gRPC. Should we have a separate CLI flag for this maybe? Also, whatever flag should also apply to our gRPC server too perhaps (not just for gRPC proxying).
> 
> We should reuse `dapr-http-max-request-size` here as this is already applied to our gRPC servers.

I see. Can we deprecate it and create `dapr-max-request-size` and use the old one as alias?

**Comment by yaron2** [comment]
> > > 1. Using `dapr-http-max-request-size` doesn't feel right for something that is related to gRPC. Should we have a separate CLI flag for this maybe? Also, whatever flag should also apply to our gRPC server too perhaps (not just for gRPC proxying).
> > 
> > 
> > We should reuse `dapr-http-max-request-size` here as this is already applied to our gRPC servers.
> 
> I see. Can we deprecate it and create `dapr-max-request-size` and use the old one as alias?

I don't see a reason to do that but we can consider a contribution (separate from this PR) in the future.

**Comment by Flern** [comment]
@yaron2 - update pushed.  I also missed some significant infrastructure updates needed to allow the `dapr-http-max-request-size` to be included in service configurations for e2e deployments.  Having trouble running e2e locally, though, to verify that this will work.

File: tests/e2e/service_invocation/service_invocation_test.go [hunk 1]
```diff
@@ -146,12 +146,13 @@ func TestMain(m *testing.M) {
...
-			Replicas:       1,
-			IngressEnabled: true,
-			MetricsEnabled: true,
+			AppName:          "grpcproxyclient",
+			DaprEnabled:      true,
+			ImageName:        "e2e-service_invocation_grpc_proxy_client",
+			Replicas:         1,
+			IngressEnabled:   true,
+			MetricsEnabled:   true,
+			MaxRequestSizeMB: 6,
```
Comment (yaron2) [inline tests/e2e/service_invocation/service_invocation_test.go:155]: You need to set `MaxRequestSizeMB` on `grpcproxyserver` as well. The YAML files under `tests/apps/` are not applied during the tests.

**Review by yaron2 — CHANGES_REQUESTED** [review]

**Comment by yaron2** [comment]
@Flern you need to add gRPC options to the proxy server app [here](https://github.com/dapr/dapr/blob/master/tests/apps/service_invocation_grpc_proxy_server/app.go#L45) for `MaxRecvMsgSize` and `MaxSendMsgSize`.

**Review by yaron2 — APPROVED** [review]

**Pull request merged**